### ISSUE TITLE
[Chore] Improve API endpoints response structure

### ISSUE THREAD

**Issue by olivierobert** [pakaponk/gsr-scraping#21]
## Issue

The API responses do not follow any JSON-compliant standard such as https://jsonapi.org/ and adequate HTTP response codes consistently. While `JwtAuthGuard` does send an error code `401` for unauthorized requests and `getScrapingJob` does use `ForbiddenException` for authorization errors, the endpoints themselves do not return adequate error responses consistently. For instance, `getKeywordFromCsv` does not use exceptions and will result in a server error, while this is no unexpected error. Or, `login` does not have any error handling if `this.authService.login` does not return a token but instead it should throw `BadRequestException`.

## Exepcted

Leverage NestJS exception handling: https://docs.nestjs.com/exception-filters#throwing-standard-exceptions

**Comment by pakaponk** [comment]
Agree. Due to the time limit, I haven't properly added appropriate [HttpExceptions](https://docs.nestjs.com/exception-filters#built-in-http-exceptions) for any anticipatable errors in all of the APIs yet.

I will start to address this issue with each API one by one. Starting from `/auth/local/register` since it also affects issue #17
